### PR TITLE
Core: Add in generic cache-manager chooser function

### PR DESCRIPTION
**Pull request by ikelos**
We've got a nice interface to use a generic cache manager, better to use that than require bits of the framework to specifically pull in our cache manager.

### CODE REVIEW

**Comment by ikelos** [comment]
I don't think this is controversial.  If it passes all the tests, I'll merge it in a couple days.

**Comment by ikelos** [comment]
Hmm, we actually want this to start a new cache if the file it's given doesn't exist.  I guess we just need to check whether the path provided could be valid?

File: volatility3/framework/automagic/symbol_cache.py [hunk 1]
```diff
@@ -492,6 +492,21 @@ def get_identifiers(self, operating_system: Optional[str]) -> List[bytes]:
...
+def load_cache_manager(cache_file: Optional[str] = None) -> CacheManagerInterface:
+    """Loads a cache manager based on a specific cache file"""
+    if cache_file is None:
+        cache_file = os.path.join(constants.CACHE_PATH, constants.IDENTIFIERS_FILENAME)
+    # Different implementations of cache
+    if not os.path.exists(cache_file):
+        raise ValueError("Non-existant cache file provided")
+    with open(cache_file, "rb") as fp:
+        header = fp.read(4)
+        if header not in [b"SQLi"]:
```
Comment (gcmoreira) [inline volatility3/framework/automagic/symbol_cache.py:504]: It's reading 4 bytes and comparing against 4 bytes, not sure why the `header not in [b"SQLi"]`? shouldn't it be `header != b"SQLi"`? or one-liner `open(cache_file, "rb").read(4) != b"SQLi"`?

**Comment by gcmoreira** [comment]
Hey @ikelos sorry about this review request, it got merged so quickly

**Comment by ikelos** [comment]
No problem, sorry I wanted to act on it while it was still fresh.  Hopefully it appears problem free?

**Comment by gcmoreira** [comment]
yeah, only that sqli header check but anyway a minor stuff

File: volatility3/framework/automagic/symbol_cache.py (hunk 1 above)
Comment (ikelos) [reply volatility3/framework/automagic/symbol_cache.py:504]: This is all for future extensibility, when we get different file formats for the cache.  The 4 byte header check will likely be used for whatever format we end up with, so I coded it with essentially a case statement with only one clause in it...

**Comment by ikelos** [comment]
Yeah, we can make it a bit longer, but I thought better in case we make a cache file that's of a different format in the future.

**Pull request merged**